apps/frontend/src/libs/components/sidebar/libs/components/sidebar-link/sidebar-link.tsx

```diff
@@ -0,0 +1,33 @@
+import { Link } from "~/libs/components/components.js";
+import { type AppRoute, SidebarIconSource } from "~/libs/enums/enums.js";
+import { type IconName, type ValueOf } from "~/libs/types/types.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	icon: IconName;
+	pathname: string;
+	title: string;
+	to: ValueOf<typeof AppRoute>;
+};
+
+const SidebarLink: React.FC<Properties> = ({
+	icon,
+	pathname,
+	title,
+	to,
+}: Properties) => {
+	const isActive = pathname === to;
+	return (
+		<Link isActive={isActive} to={to} type="navLink">
+			<img
+				alt={title}
+				className={styles["linkIcon"]}
+				src={SidebarIconSource[icon]}
```

Comment:
Create a new Icon component, which will be used like this:
`<Icon iconName="name" />`
No need to have SidebarIconSource

Reply:
this Icon component will have all of the URIs to svgs inside of it? Or how should it be implemented otherwise?

Reply:
It still needs to be some sort of a map to retrieve the image url by name, or am I missing something?

Reply:
Maybe template string?

Reply:
We will have a map `iconNameToSvg` from where we will get the required icon. That will be a part of Icon component
All icon svgs will be imported to that map file

Reply:
Will this map be a separate task or should I implement it right now?

Reply:
You should implement it now - add only the icons that you use for this task